### ISSUE TITLE
Usage of get_beam_attrs_along_segment does not keep computational graph intact

### ISSUE THREAD

**Issue by phys-cgarnier** [desy-ml/cheetah#525]
While using the method from segment.py : get_beam_attrs_along_segment to aquire beam attributes, the tensors returned do not keep the computational graph intact. These parameters can be used directly in the loss function and still keep its differentiability with respect to the original parameters intact.

**Comment by phys-cgarnier** [comment]
@jank324 Sorry about the typo I fixed it.
Here is my training loop. I don't know if this is an issue or more of a feature request,  but, changing the quad misalignment should change the beam position. When I grab the position of the beam for some elements using the method .get_beam_attrs_along_segment I cannot use those values in my loss function directly and keep the computational graph intact.
        


#dl/dquadmis = dl/dsim_bpm_mux * dsim_bpm_mux/dquadmis 

def train(num_steps: int, lr=.05) -> dict:
    device_settings = []
    loss_history = []
    sub_segment.qdg008.misalignment = nn.Parameter(sub_segment.qdg008.misalignment)
    sub_segment.qdg009.misalignment = nn.Parameter(sub_segment.qdg009.misalignment)
    sub_segment.qdg010.misalignment = nn.Parameter(sub_segment.qdg010.misalignment)
    sub_segment.qdg011.misalignment = nn.Parameter(sub_segment.qdg011.misalignment)
    for name, param in sub_segment.parameters():
        print(name, param.requires_grad)


    optimizer = torch.optim.Adam(sub_segment.parameters(), lr=lr)
    for _ in range(num_steps):
        optimizer.zero_grad()
        sub_segment.tcxdg0.phase = (phases - 45)/360
        sub_segment.track(beam)
        position_data = sub_segment.get_beam_attrs_along_segment(("mu_x"),incoming=beam)
        sim_bpm_mux = position_data[...,ind]
        pred = sim_bpm_mux.T * 1000
        target = slice_exp_bpms
        loss = F.mse_loss(pred,target)

        #is non differentiable
        loss.backward()

        
        optimizer.step()
        loss_history.append(loss.item())         
        device_settings.append(
            torch.stack(
                [
                    sub_segment.qdg008.misalignment,
                    sub_segment.qdg009.misalignment,
                    sub_segment.qdg010.misalignment,
                    sub_segment.qdg011.misalignment,
                ]
            )
            .detach()
            .numpy()
        )
    history = {
        "loss": loss_history,
        "device_settings": device_settings,
    }
    return history

**Comment by roussel-ryan** [comment]
@phys-cgarnier this example cannot be reproduced / run by @jank324 since he doesn't have the DIAG0 lattice file. A minimal example would be a basic quad + drift or similar and a demonstration that the gradient of the output doesn't exist. Here is an example
```
import cheetah

k1 = torch.tensor(1.0,requires_grad=True)
segment = cheetah.Segment(
        elements=[
            cheetah.Quadrupole(length=torch.tensor(0.1), k1=k1)
            cheetah.Drift(length=torch.tensor(1.0)),
        ],
    )
data = segment.get_beam_attrs_along_segment(("mu_x"),incoming=beam)
data.backward()

assert k1.grad is not None
